pkg/app/run.go

```diff
@@ -149,7 +151,23 @@ func Run(opts RunOpts) error {
 		}
 	}
 
-	globals.APIClient, err = opts.APIClient(token, endpoint)
+	if auth.Required(name, token, endpoint, tSource, opts.Stdout, opts.ClientFactory) {
+		token, err = auth.Init(opts.Stdin, opts.Stdout, auth.Browser, opts.AuthService)
+		if err != nil {
+			return err
+		}
+		err = auth.PersistToken(token, globals.Path, endpoint, &globals.File, opts.ClientFactory)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Here is where we add calls out to the new authentication logic.

If `auth.Required` returns true, then we'll start a new authentication flow (and persist the newly generated token back to the user's local CLI configuration file).  

Otherwise we'll continue on executing the command using the existing token we have.